lib/url.js

```diff
@@ -353,6 +353,13 @@ function urlFormat(obj) {
   // this way, you can call url_format() on strings
   // to clean up potentially wonky urls.
   if (typeof obj === 'string') obj = urlParse(obj);
+
+  if (typeof obj !== 'object')
```

Comment:
V8 has pattern-matching logic for `if (typeof(value) === 'type')` guards.  Storing the result of `typeof(value)` in a local before comparing kills that.